src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -745,7 +745,8 @@ public String[] getTruststoreJavaOptions() {
     /**
      * One step to run.
      * <p>Since this thunk will be sent to a different JVM, it must be serializable.
-     * The test class will certainly not be serializable, so you cannot use an anonymous inner class.
+     * The test class will certainly not be serializable, so you cannot use an anonymous inner class,
+     * and regular lambdas also risk accidentally capturing non-serializable objects from scope.
```

Comment:
Admittedly you will find that out at runtime with a `NotSerializableException`. My bigger concern is that the serial form of a general lambda (not a method handle) might not carry over from the test JVM to the Jenkins JVM, especially if in the future we support running the Jenkins JVM in a container that might be using a distinct version of Java. Maybe I should stop worrying and trust that https://docs.oracle.com/javase/8/docs/api/serialized-form.html#java.lang.invoke.SerializedLambda will be reliable? The bytecode sent to the Jenkins JVM is guaranteed to be identical to that run in the test JVM, so there is no concern about choice of compiler, version skew as test code is edited over time, etc.; [this post](https://cr.openjdk.org/~briangoetz/eg-attachments/lambda-serialization.html) implies that there is no risk for a given bytecode version.

Reply:
⇒ #882